torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -0,0 +1,93 @@
+#include "torch/csrc/jit/passes/constant_propagation.h"
+#include "torch/csrc/jit/passes/dead_code_elimination.h"
+#include "torch/csrc/utils/functional.h"
+#include "torch/csrc/jit/interpreter.h"
+
+namespace torch { namespace jit {
+
+std::unordered_set<Symbol> skip_list = {
+  //FIXME If & Loop require special casing because they cannot be run as a
+  //single node.
+  prim::If,
+  prim::Loop,
+  //FIXME Same problem as in DCE - cpp & python PythonOp and CppOp should be
+  //FIXME treated as having side effects but ONNX depends on them being removed
+  prim::Print,
+ };
+
+std::vector<at::Tensor> runNode(Node *n) {
+  auto graph = std::make_shared<Graph>();
+  auto block = graph->block();
+  //very similar to Block::cloneFrom, but here we are cloning just the node,
+  //and not recursively
+  std::unordered_map<Value*, Value*> local_map;
+  auto env = [&](Value * v) {
+    auto it = local_map.find(v);
+    if(it != local_map.end())
+      return it->second;
+    barf("Encountered a use of a value not in scope");
+  };
+  for(auto input : n->inputs()) {
+    local_map[input] = block->addInput()->copyMetadata(input)->setStage(input->stage());
+    graph->setStage(std::max(graph->stage(), input->stage()));
+  }
+  auto new_node = block->appendNode(graph->createClone(n, env, /*non-recursive */ false));
```

Comment:
Why non-recursive? Do we ever want to run nodes with blocks in here? Maybe it's better to just assert it doesn't have any?

Reply:
Good point. Now it's special-casing Loop & IF I think we can assert that there are no blocks.